source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -0,0 +1,155 @@
+
+Hooks are a list of places in the Open edX platform where externally defined functions can take place. These functions may alter what the user sees or experiences on the platform, while in other cases, they act as notifications. All hooks are designed to be extended through `Open edX Django plugins`_ and configurations.
+
+Hooks can be of two types: events and filters. Events are signals sent in specific places whose receivers can extend functionality, while filters are functions that can modify the application's behavior. To allow extension developers to use these definitions in their implementations, both kinds of hooks are defined in lightweight external libraries:
+
+* `openedx-filters`_
+* `openedx-events`_
+
+The main goal of the framework is that developers can use it to change the platform's functionality as needed and still migrate to newer Open edX releases with little to no development effort. So, the framework is designed with stability in mind, meaning it is versioned and backward compatible as much as possible.
+
+A longer description of the framework and its history can be found in `OEP 50`_.
+
+.. _OEP 50: https://open-edx-proposals.readthedocs.io/en/latest/oep-0050-hooks-extension-framework.html
+.. _openedx-filters: https://github.com/openedx/openedx-filters
+.. _openedx-events: https://github.com/openedx/openedx-events
+.. _open-closed principle: https://docs.openedx.org/projects/edx-platform/en/open-release-quince.master/concepts/extension_points.html
+
+Why adopt the Hooks Extension Framework?
+========================================
+
+Stable and Maintainable Extensions
+----------------------------------
+
+The Hooks Extension Framework allows developers to extend the platform's functionality in a stable, maintainable, and decoupled way ensuring easier upgrades and long-term stability by removing the need to modify the core in an significant way.
+
+Contained Solution Implementation
+---------------------------------
+
+By avoiding core modifications, the framework promotes self-contained solutions, eliminating the need for custom code to coexist with core logic which lowers maintenance costs for extension developers.
+
+Leveraging the Open edX Plugin Extension Mechanism
+--------------------------------------------------
+
+The framework allows developers to implement custom business logic and integrations directly in plugins. This keeps core modifications minimal, focusing maintenance and development efforts on plugins, where solutions can be built and maintained independently of the core platform.
+
+Standardization
+---------------
+
+Both filters and events implementations implement an approach for adding additional features, such as communication between services or backend flow control. With these standards in place, it’s easy to identify when and how to use the framework as a solution, ensuring a consistent and predictable approach to extending the platform.
```

Comment:
```suggestion
The implementation of both filters and events allows for adding additional features, such as communication between services or backend flow control.
```

I don't really understand what the second sentence means (" With these standards in place, it’s easy to identify when and how to use the framework as a solution, ensuring a consistent and predictable approach to extending the platform.") - could you explain what you're trying to convey, so I can suggest a rewording? I'm particularly unsure of what "these standards" means.

Reply:
Events and filters solve two problems: 1. Communication and 2. Application behavior modifications, respectively. So, if a problem can be solved by one of the two, then developers can use events/filters. I think this gives developers a standard to follow for extensibility if the use case fits what I previously mentioned. Does this make sense?

Reply:
Events and filters are used for 1. Communication and 2. Application behavior modifications, respectively. So, if a problem can be solved by one of those two options, then the implementation would be pretty straightforward—at least the extension part of it. This would give developers a standard to follow for extensibility if the use case fits what I previously mentioned. Does this make sense?

Reply:
What do you think about using this instead?

> Filters and events provide developers with a standard for adding new features to the platform through extension mechanisms. Events primarily handle communication or synchronization between different parts of the application, while filters modify application behavior. If a problem can be addressed through one of these two options, the implementation becomes pretty straightforward. 

I'm not quite sure if "Standardization" is the best title for this, though.

Reply:
@sarina: what do you think about my latest suggestion?